### ISSUE TITLE
[wdspect] webdriver/tests/bidi/script/call_function/invalid.py times out with the "harness TIMEOUT" error

### ISSUE THREAD

**Issue by OrKoN** [web-platform-tests/wpt#42220]
See https://wpt.fyi/results/webdriver/tests/bidi/script/call_function?sha=e29a79361e&label=master&max-count=1

The same test passes with Firefox and it passes when running WPT locally.

cc @whimboo @sadym-chromium @thiagowfx

**Comment by whimboo** [comment]
It might be good to reference a complete log which might indicate if there is one or more tests that are running longer and cause this test timeout.

**Comment by OrKoN** [comment]
@whimboo where can I find a link to the complete log? (the wpt.fyi link is tied to a specific run on the main branch but I don't see any links to detailed logs)

**Comment by OrKoN** [comment]
Looks like the following is the detailed log https://community-tc.services.mozilla.com/tasks/dJgWYCj3RR-vrARDdBJKUw/runs/0/logs/live/public/logs/live.log

**Comment by OrKoN** [comment]
There are errors like:

```
 1:19.42 pid:2239 [2520:2520:0928/151843.159152:WARNING:chrome_main_delegate.cc(613)] This is Chrome version 118.0.5993.11 (not a warning)
 1:19.49 pid:2239 [2520:2546:0928/151843.231751:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /var/run/dbus/system_bus_socket: No such file or directory
 1:19.49 pid:2239 [2520:2546:0928/151843.231800:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /var/run/dbus/system_bus_socket: No such file or directory
 1:19.49 pid:2239 [2520:2543:0928/151843.235310:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /var/run/dbus/system_bus_socket: No such file or directory
 1:19.49 pid:2239 [2520:2543:0928/151843.235355:ERROR:bus.cc(407)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
 1:19.49 pid:2239 [2520:2543:0928/151843.235380:ERROR:bus.cc(407)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
 1:19.49 pid:2239 [2520:2543:0928/151843.235389:ERROR:bus.cc(407)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
 1:19.49 pid:2239 [2520:2543:0928/151843.235395:ERROR:bus.cc(407)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
 ```
 
 in the logs which indicates that the host does not have dbus or permissions for dbus are not correct.

**Comment by whimboo** [comment]
I don't think that this is the problem. The timestamps are all coming from the same second. Lets check:

1. The test starts at [0:25:45](https://community-tc.services.mozilla.com/tasks/dJgWYCj3RR-vrARDdBJKUw/runs/0/logs/live/public/logs/live.log#L373)
2. DevTools are listening at [0:27:04](https://community-tc.services.mozilla.com/tasks/dJgWYCj3RR-vrARDdBJKUw/runs/0/logs/live/public/logs/live.log#L397)
3. The first test is done at [0:27.49](https://community-tc.services.mozilla.com/tasks/dJgWYCj3RR-vrARDdBJKUw/runs/0/logs/live/public/logs/live.log#L408) which is about 2s later
4. Now each test takes around 30-40ms
5. The last runnable test before the timeout finishes at [0:55:45](https://community-tc.services.mozilla.com/tasks/dJgWYCj3RR-vrARDdBJKUw/runs/0/logs/live/public/logs/live.log#L844)

When you now compare to Firefox [each test only takes 3ms](https://community-tc.services.mozilla.com/tasks/c-l9Xg8nQV2PrmdnBnijKw/runs/0/logs/live/public/logs/live.log#L458) which is a factor of 10 faster.

**Comment by OrKoN** [comment]
@whimboo I would not rule out the dbus issue as it might make things run slower.

**Comment by OrKoN** [comment]
there seems to also some background component update running that should probably be disabled for testing. The version on Linux for the Dev channel also seems to be somewhat outdated (although that should not make a difference).

**Comment by whimboo** [comment]
Hm, in the logs for Firefox the dbus issue cannot be seen. I assume that for Chrome a different worker setup is used?

**Comment by OrKoN** [comment]
@whimboo I am not 100% sure why Chrome attempts to use dbus as I thought it is mainly a ChromeOS thing. We get similar reports in Puppeteer when people try to run Chrome in Docker: oftentimes it is connected to timeouts and slowness.

**Comment by rakuco** [comment]
I've tried to see if there was anything wrong related to D-Bus or something else, but it just looks like there are too many tests in the file and it needs the extra metadata from #42356.

I reverted the PR above and used `./wpt docker-build` with different `Dockerfile`s using "ubuntu-20.04", "ubuntu-22.04", "python:3.8", "python:3.9" and "python:3.10" as base images. After that I ran `./wpt run --log-mach=- chrome webdriver/tests/bidi/script/call_function/invalid.py --headless --log-mach verbose --install-browser -y` without even starting an X server (and therefore getting even more X and D-Bus related errors than usual) and the tests always time out after running for ~30 seconds.

I've also run the same command on my regular Linux desktop setup (with and without `--headless`) and got the same timeouts.

That [`WdspecTest.default_timeout`](https://github.com/web-platform-tests/wpt/blob/28cb182732bc42d1fb685f06906a3293726c9d40/tools/wptrunner/wptrunner/wpttest.py#L699) is 25 and `WdspecExecutor` [calculates](https://github.com/web-platform-tests/wpt/blob/28cb182732bc42d1fb685f06906a3293726c9d40/tools/wptrunner/wptrunner/executors/base.py#L639) its `timeout` value to 30 in this case, the timeout is expected.
